packages/trader/src/Modules/Trading/Containers/trade-chart.tsx

```diff
@@ -0,0 +1,199 @@
+            chartType={chart_type}
+            clearChart={false}
+            contracts_array={markers_array}
+            crosshair={is_mobile ? 0 : undefined}
+            crosshairTooltipLeftAllow={560}
+            feedCall={{
+                activeSymbols: false,
+            }}
+            enabledNavigationWidget={isDesktop()}
+            enabledChartFooter={false}
+            getMarketsOrder={getMarketsOrder}
+            granularity={show_digits_stats || is_accumulator ? 0 : Number(granularity)}
+            hasAlternativeSource={has_alternative_source}
+            id='trade'
+            importedLayout={chart_layout}
+            initialData={{
+                activeSymbols: JSON.parse(JSON.stringify(active_symbols)),
+            }}
+            isConnectionOpened={is_socket_opened}
+            isLive
+            isMobile={is_mobile}
+            is_beta={is_beta_chart}
+            leftMargin={isDesktop() && is_positions_drawer_on ? 328 : 80}
+            maxTick={is_mobile ? max_ticks : undefined}
+            onExportLayout={exportLayout}
+            requestAPI={wsSendRequest}
+            requestForget={wsForget}
+            requestForgetStream={wsForgetStream}
+            requestSubscribe={wsSubscribe}
+            settings={settings}
+            shouldFetchTradingTimes
+            should_show_eu_content={should_show_eu_content}
+            should_zoom_out_on_yaxis={is_accumulator}
+            showLastDigitStats={isDesktop() ? show_digits_stats : false}
+            stateChangeListener={chartStateChange}
+            symbol={symbol}
+            toolbarWidget={() => {
+                if (is_beta_chart) {
+                    return (
+                        <ToolbarWidgetsBeta
+                            is_mobile={is_mobile}
+                            updateChartType={updateChartType}
+                            updateGranularity={updateGranularity}
+                        />
+                    );
+                }
+                return (
+                    <ToolbarWidgets
+                        is_mobile={is_mobile}
+                        updateChartType={updateChartType}
+                        updateGranularity={updateGranularity}
+                    />
+                );
+            }}
+            topWidgets={is_trade_enabled ? topWidgets : undefined}
+            yAxisMargin={{
+                top: is_mobile ? 76 : 106,
+            }}
+        >
+            {!is_beta_chart &&
+                markers_array.map(marker => {
+                    const Marker = AllMarkers[marker.type as keyof typeof AllMarkers];
+                    return (
+                        <Marker
+                            currency={currency}
+                            granularity={granularity}
+                            is_dark_theme={is_dark_mode_on}
+                            {...marker}
+                            key={marker.key}
```

Comment:
If `key` already exist is in `marker` object, why do you rewriting it? Sorry, if it's not relevant

Reply:
@kate-deriv yes, it's inside the object but if you try to remove it, sadly you'll get the TS error about the absence of the key immediately.